src/utils/removeMessageReservedFields.ts

```diff
@@ -0,0 +1,52 @@
+import type { MessageType } from '../components/MessageList/hooks/useMessageList';
```

Comment:
I think this can be reduced to:

```
const reserved = [
  'cid',
  'config',
  'created_at',
  'created_by',
  'id',
  'last_message_at',
  'member_count',
  'type',
  'updated_at',
] as const;

type ReservedType = typeof reserved;

export const removeReservedFields = <T extends Record<string, unknown>>(
  message: T,
) => {
  const retryMessage = { ...message };

  reserved.forEach((key) => {
    delete retryMessage[key];
  });

  return retryMessage as Omit<T, ReservedType[number]>;
};
```